### PR TITLE
add updated version of py-dnaio

### PR DESCRIPTION
**Pull request by glennpj**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-dnaio/package.py [hunk 1]
```diff
@@ -13,6 +13,7 @@ class PyDnaio(PythonPackage):
     url      = "https://pypi.io/packages/source/d/dnaio/dnaio-0.3.tar.gz"
     git      = "https://github.com/marcelm/dnaio.git"
 
+    version('0.4.2', sha256='fa55a45bfd5d9272409b714158fb3a7de5dceac1034a0af84502c7f503ee84f8')
     version('0.3', sha256='47e4449affad0981978fe986684fc0d9c39736f05a157f6cf80e54dae0a92638')
 
     depends_on('python@3.4:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-dnaio/package.py:19]: Can you add a build dep on `py-setuptools-scm`?

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-dnaio/package.py (hunk 1 above)
Comment (glennpj) [reply var/spack/repos/builtin/packages/py-dnaio/package.py:19]: Yes, just added a commit.

File: var/spack/repos/builtin/packages/py-dnaio/package.py [hunk 2]
```diff
@@ -13,8 +13,11 @@ class PyDnaio(PythonPackage):
     url      = "https://pypi.io/packages/source/d/dnaio/dnaio-0.3.tar.gz"
     git      = "https://github.com/marcelm/dnaio.git"
 
+    version('0.4.2', sha256='fa55a45bfd5d9272409b714158fb3a7de5dceac1034a0af84502c7f503ee84f8')
     version('0.3', sha256='47e4449affad0981978fe986684fc0d9c39736f05a157f6cf80e54dae0a92638')
 
-    depends_on('python@3.4:', type=('build', 'run'))
+    depends_on('python@3.5:', when='@0.4.2:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-dnaio/package.py]: I don't think this is actually true is it? I think 3.4+ is fine.

Comment (glennpj) [reply var/spack/repos/builtin/packages/py-dnaio/package.py]: Argh, I was looking at the file from the master branch. I will back that out.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**